### DISCUSSION TITLE
How to get update trigger in JWT callback when using Credentials provider by using useSession().update().

### DISCUSSION THREAD

**Discussion by mjulstein** [nextauthjs/next-auth#12214] [Help]
Hi. I have this simplified JWT callback in a credentials provider:
```TypeScript
async jwt({ token, user, trigger, account, profile, session, ...more }) {
      if (trigger === 'signIn') {
       // I get here every time I sign in.
        token.myStuff = getMyStuff();
      } else if (trigger === 'update') {
        // I expect to get here when using useSession().update();
       token.myStuff = getMyStuff(); // refresh the stuff (after user updated their profile)
      }
      return token; // I get here on useSession().update(), but trigger is undefined;
    },
```
I have also tired passing an empty object (`{}`) to update(), but then the backend throws and the session token gets destroyed.

I haven't found any place where this behavior is documented.
I would rather just fetch the user profile when something has changed rather than every time JWT is called. Is there a different way I can refresh the session?

This discussion seems similar, but it also seems quite stale: https://github.com/nextauthjs/next-auth/discussions/3941#discussioncomment-11125876

**Comment by mjulstein** [comment]
For some reason now it works with using `useSession().update({})`
Could it be that the update function I was calling was stale for some reason?
The first time I tried to use it throwed and exception and destroyed the session cookie, but now it works.

**Comment by maxpaleo** [comment]
What version are you on? 5.0.0-beta.25 , does not trigger `update()`from `useSession()` for me, no matter what I try. The docs literally don't mention anything about it either.

**Reply by Abhishek21k** [reply]
facing same issue, also we can't call the update() trigger from the server side?

**Comment by Abhishek21k** [comment]
did anyone found any soln to this/

**Comment by Abhishek21k** [comment]
in the auth.js file we need to give the session as the argument on the useSession().update({session})
```
callbacks: {
  jwt: ({ token, user, trigger, session }) => {
      if (trigger === "update") {
        token.role = session.user.role;
      }
      if (user) {
        token.role = user.role;
        token.emailVerified = user.emailVerified;
      }
      return token;
    },
}

 ```
 
 from the client side it will look this:
 ```
"use client";
import { useSession } from "next-auth/react";

 const session = useSession();

 await session.update({
          ...session.data,
          user: {
            ...session.data.user,
            role: "ADMIN",
          },
        });
```

on the jwt callback session argument hover over it it says that it need to be given when triggering an update:
<img width="996" alt="Screenshot 2025-01-11 at 8 09 31 PM" src="https://github.com/user-attachments/assets/5405b1e3-8abf-455d-ba6a-c672f0c9677b" />
